REVIEW_COPY.md

```diff
@@ -0,0 +1,312 @@
+
+Lastly the drag stop event. We already changed the outline while the user was moving the cursor, and so all that's left to do here is change the model to its non-dragging state by setting `draggedNode` to `Nothing`.
+
+### Moving nodes in an outline
+
+Our hardest task is figuring out where to put our dragged node when the user moves the cursor. What might the user intent when dragging the cursor to a certain location? To move the dragged node in front of another node, behind it, or nested beneath it?
+
+Using dropzones we could draw invisible boxes in those positions that activate when the user moves over them, but the experience is unlikely to be great. Make the boxes too small and the user will not spend a lot of time over them, making the interface unresponsive for most of the time. Make the boxes too big and they start to overlap, causing the upper box instead of the closest one that will receive the dragged node.
+
+Lets forget about dropzones and think about the behavior we want. As the user moves, we'd like to display the candidate node in the location closest to the cursor position that might receive it. To be able to figure out which of those locations is closest we need to know where they are. To do this we are going to put invisible elements in the DOM at each location where we can insert the dragged node. Contrary to the dropzones approach we're not going to bother giving these elements any special dimensions or positioning. We want them to just flow with the content on the page, and keep us apprised of their location. These aren't dropzones but beacons.
+
+Apart from their coordinates in the DOM, our beacons will also need to describe their location in the outline. A beacon can define its location in relative to another node in the outline.
+
+```elm
+type CandidatePosition
+    = Before String
+    | After String
+    | PrependedIn String
+    | AppendedIn String
+```
+
+We'll create a JSON encoder for this type so we can tag each beacon element with a data attribute containing its position in the outline. We'll then set up our JavaScript to find all elements with such a data attribute in the DOM and feed their coordinates back to us on each drag event. That will allow us to define a type for drag events containing the positions of our beacons on the screen.
+
+```elm
+type alias DragMsg =
+    { cursor : Coords
+    , beacons : List Beacon
+    }
+
+
+type alias Beacon =
+    ( CandidatePosition, Rect )
+```
+
+Remember the `closestRect` function from the polygon example? It's what we need to find the `CandidatePosition` closest to the cursor! Once we know which candidate position is closest, all we need is a function to that moves a node to its new position in an outline:
+
+```elm
+moveNodeInOutline : String -> CandidatePosition -> List OutlineNode -> List OutlineNode
+moveNodeInOutline = Debug.todo "See full source code for implementation!"
```

Comment:
if you're not going to show the source, I don't think you need a code block here. Tell folks to go look at the full source or include and annotate it.